### ISSUE TITLE
`wildcard_enum_match_arm` lint don't work in case of a single variant

### ISSUE THREAD

**Issue by mhnap** [rust-lang/rust-clippy#12115]
### Summary

I tried to use an example from `wildcard_enum_match_arm` [lint doc](https://rust-lang.github.io/rust-clippy/master/index.html#/wildcard_enum_match_arm), but it didn't work. I also noticed that there is a separate `match_wildcard_for_single_variants` [lint for a single variant](https://rust-lang.github.io/rust-clippy/master/index.html#/match_wildcard_for_single_variants), but if it's expected to use it instead then example seems need to be changed.

### Lint Name

wildcard_enum_match_arm

### Reproducer

I tried this code: ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cba16e106f8d46965760e8c1a9b12ea8))

```rust
#![warn(clippy::wildcard_enum_match_arm)]

fn main() {
    enum Foo {
        A(usize),
        B(usize),
    }

    let x = Foo::B(1);

    match x {
        Foo::A(_) => {}
        _ => {}
    }
}

```

I expected to see this happen: Lint warning emitted.

Instead, this happened: No lint warning.

### Version

```text
rustc 1.74.1 (a28077b28 2023-12-04)
binary: rustc
commit-hash: a28077b28a02b92985b3a3faecf92813155f1ea1
commit-date: 2023-12-04
host: x86_64-unknown-linux-gnu
release: 1.74.1
LLVM version: 17.0.4
```
